proposals/csharp-9.0/unconstrained-type-parameter-annotations.md

```diff
@@ -0,0 +1,67 @@
+﻿# Unconstrained type parameter annotations
+
+## Summary
+
+Allow nullable annotations for type parameters that are not constrained to value types or reference types: `T?`.
+```C#
+static T? FirstOrDefault<T>(this IEnumerable<T> collection) { ... }
+```
+
+## `?` annotation
+
+In C#8, `?` annotations could only be applied to type parameters that were explicitly constrained to value types or reference types.
+In C#9, `?` annotations can be applied to any type parameter, regardless of constraints.
+
+If a type parameter `T` is substituted with a reference type, then `T?` represents a nullable instance of that reference type.
+If `T` is substituted with a value type, then `T?` is represents an instance of `T`.
```

Comment:
What if `T` is a nullable value type?

Reply:
@pkulikov do you mean what if `T` is substituted with a nullable value type (say `int?`)? In that case then the result is `int?`. Essentially in the case where the type argument is a value type then the `?` is effectively dropped from the type parameter.

Reply:
> do you mean what if `T` is substituted with a nullable value type (say `int?`)? In that case then the result is `int?`

@jaredpar yes, thanks!

> Essentially in the case where the type argument is a value type then the `?` is effectively dropped from the type parameter.

That's a good clarification, thank you.